### PR TITLE
Added FlowContext extraction rules and passwords encryption

### PR DESCRIPTION
**Pull request by tobiasake**
2.0.0-maintainence branch was used for doing the implementation of adding FlowContext rules and encryption passsword property handling that needed to be added and based on the 2.0.0 release, due to we have committed  new features on master branch that can not yet be released.
2.0.0-maintainence branch was branched out from 2.0.0 release and has the FlowContext rules and the encryption handling of password properties. This has been released as version 2.0.1 from this branch.
This pull-request is for merging these 2.0.1 version changes to the master branch.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/java/com/ericsson/ei/config/ConfigurationLogger.java [hunk 1]
```diff
@@ -57,8 +57,8 @@ private void logConfiguration() {
                 + "threads.corePoolSize: " + environment.getProperty("threads.corePoolSize") + "\n"
                 + "threads.queueCapacity: " + environment.getProperty("threads.queueCapacity") + "\n"
                 + "threads.maxPoolSize: " + environment.getProperty("threads.maxPoolSize") + "\n"
-                + "failed.notification.collection-name: " + environment.getProperty("failed.notification.collection-name") + "\n"
-                + "failed.notification.database-name: " + environment.getProperty("failed.notification.database-name") + "\n"
+                + "missedNotificationCollectionName: " + environment.getProperty("missedNotificationCollectionName") + "\n"
+                + "missedNotificationDataBaseName: " + environment.getProperty("missedNotificationDataBaseName") + "\n"
```
Comment (e-pettersson-ericsson) [inline src/main/java/com/ericsson/ei/config/ConfigurationLogger.java]: I don't know what happened here in the rebase, but on master we want to have the name change to failed notifications right?

File: wiki/markdown/configuration.md [hunk 2]
```diff
@@ -168,4 +168,57 @@ and the settings are defined in the **ldap.server.list** property.
...
+
+These Eiffel-Intelligence password properties can be provided encrypted:
+        
+        rabbitmq.password
+        spring.data.mongodb.password
+
+
+A password property is configured by providing the encrypted password in this format.
+
+        some.component.password=ENC(<encrypted password>)
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Does mvn site like the < signs? Have you checked locally if the mvn site goal can render HTML pages from this file? If not, replace the < with { characters instead.

File: wiki/markdown/configuration.md [hunk 3]
```diff
@@ -168,4 +168,57 @@ and the settings are defined in the **ldap.server.list** property.
...
+ 
+    ldap.server.list=[{\
+        "url": "<ldap address>",\
+        "base.dn": "<dn>",\
+        "username": "<username>",\
+        "password": "ENC(<encrypted password>)",\
+        "user.filter": "<filter>"\
+    }]
+
+How and examples of how to encrypt passwords can be found on Jasypt homepage.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Remove the first "How and" in the beginning of the sentence.

File: wiki/markdown/configuration.md [hunk 4]
```diff
@@ -168,4 +168,57 @@ and the settings are defined in the **ldap.server.list** property.
...
+On Jasypt homepage encrypt.sh script can be downloaded, which can be used to encrypt passwords:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Should we add a link to Jasypts webpage here?

File: src/main/java/com/ericsson/ei/config/ConfigurationLogger.java (hunk 1 above)
Comment (tobiasake) [reply src/main/java/com/ericsson/ei/config/ConfigurationLogger.java]: done.

File: wiki/markdown/configuration.md (hunk 4 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: the link is below this line.

File: wiki/markdown/configuration.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: Done.

File: wiki/markdown/configuration.md (hunk 2 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: Solved.

File: src/main/java/com/ericsson/ei/EndpointSecurity.java [hunk 5]
```diff
@@ -67,20 +73,35 @@ protected void configure(HttpSecurity http) throws Exception {
...
-            JSONObject server = (JSONObject) serverList.get(i);
-            auth.eraseCredentials(false)
-                .ldapAuthentication()
-                .userSearchFilter(server.getString("user.filter"))
-                .contextSource()
+private void addLDAPServersFromList(JSONArray serverList, AuthenticationManagerBuilder auth) throws Exception {
+    TextFormatter textFormatter = new TextFormatter();
+    StandardPBEStringEncryptor encryptor = new StandardPBEStringEncryptor();
+
+    encryptor.setPassword(jasyptEncryptorPassword);
```
Comment (e-pettersson-ericsson) [inline src/main/java/com/ericsson/ei/EndpointSecurity.java]: What happens if jasyptEncryptorPassword is not set in application.properties?

**Pull request closed without merging**